File: socialNetwork/src/main/java/com/ftn/socialNetwork/model/entity/Banned.java
package com.ftn.socialNetwork.model.entity;

import javax.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "banned")
public class Banned {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, columnDefinition = "DATETIME DEFAULT CURRENT_TIMESTAMP")
    private LocalDate timeStamp;

    @Column(nullable = false, columnDefinition = "boolean default false")
    private boolean isBlocked;
    @ManyToOne
    private GroupAdmin groupAdmin;
    @ManyToOne
    private Group group;
    @ManyToOne
    private  User bannedUser;
    @ManyToOne
    private  User user;
}

File: src/app/models/banned.ts
import { Group } from "./group";
import { GroupAdmin } from "./groupAdmin";
import { User } from "./user";

export class Banned {
    id: number;
    timeStamp: Date;
    isBlocked: boolean;
    groupAdmin: GroupAdmin | null;
    group: Group | null;
    user: User;
    bannedUser : User | null;
  
    constructor(
      id: number,
      timeStamp: Date,
      isBlocked: boolean,
      groupAdmin: GroupAdmin | null,
      group: Group | null,
      user: User,
      bannedUser : User | null
    ) {
      this.id = id;
      this.isBlocked = isBlocked;
      this.timeStamp = timeStamp;
      this.groupAdmin = groupAdmin;
      this.group = group;
      this.user = user;
      this.bannedUser = bannedUser;
    }
  }
  